Services/Init/classes/class.ilErrorHandling.php

```diff
@@ -410,13 +410,37 @@ protected function devmodeHandler() {
 			case "PLAIN_TEXT":
 				return new ilPlainTextHandler();
 			case "PRETTY_PAGE":
-				return new PrettyPageHandler();
+				// fallthrough
 			default:
-				if ($ilLog) {
-					$ilLog->write("Unknown or undefined error handler '".ERROR_HANDLER."'. "
-								 ."Falling back to PrettyPageHandler.");
+				if ((!defined('ERROR_HANDLER') || ERROR_HANDLER != 'PRETTY_PAGE') && $ilLog) {
+					$ilLog->write(
+						"Unknown or undefined error handler '".ERROR_HANDLER."'. " .
+						"Falling back to PrettyPageHandler."
+					);
+				}
+
+				$prettyPageHandler = new PrettyPageHandler();
+				if (defined('ERROR_EDITOR_URL')) {
+					$pathTranslations = [];
+
+					if (defined('ERROR_EDITOR_PATH_TRANSLATIONS')) {
+						$mappings = explode('|', ERROR_EDITOR_PATH_TRANSLATIONS);
+						foreach ($mappings as $mapping) {
+							$parts = explode(',', $mapping);
+							$pathTranslations[$parts[0]] = $parts[1];
+						}
+					}
+
+					$prettyPageHandler->setEditor(function ($file, $line) use ($pathTranslations) {
+						foreach ($pathTranslations as $from => $to) {
+							$file = preg_replace('@' . $from . '@' , $to, $file);
+						}
```

Comment:
Maybe factor this out into `applyPathMappings` or similar, this was hard to understand at first.